public/app/features/provisioning/dashboardLoader.ts

```diff
@@ -0,0 +1,63 @@
+import { getBackendSrv } from '@grafana/runtime';
+import { DashboardDTO } from 'app/types';
+
+import { AnnoKeyManagerIdentity, AnnoKeyManagerKind, AnnoKeySourcePath } from '../apiserver/types';
+
+import { BASE_URL } from './api/baseAPI';
+
+/**
+ *
+ * Load a dashboard from repository
+ */
+export async function loadDashboardFromProvisioning(repo: string, path: string): Promise<DashboardDTO> {
+  const params = new URLSearchParams(window.location.search);
+  const ref = params.get('ref'); // commit hash or branch
+
+  const url = `${BASE_URL}/repositories/${repo}/files/${path}`;
+  return getBackendSrv()
+    .get(url, ref ? { ref } : undefined)
+    .then((v) => {
+      // Load the results from dryRun
+      const dryRun = v.resource.dryRun;
+      if (!dryRun) {
+        return Promise.reject('failed to read provisioned dashboard');
+      }
+
+      if (!dryRun.apiVersion.startsWith('dashboard.grafana.app')) {
+        return Promise.reject('unexpected resource type: ' + dryRun.apiVersion);
+      }
+
+      // Make sure the annotation key exists
+      let anno = dryRun.metadata.annotations;
+      if (!anno) {
+        dryRun.metadata.annotations = {};
+      }
+      anno[AnnoKeyManagerKind] = 'repo';
+      anno[AnnoKeyManagerIdentity] = repo;
+      anno[AnnoKeySourcePath] = ref ? path + '#' + ref : path;
+
+      return {
+        meta: {
+          canStar: false,
+          isSnapshot: false,
+          canShare: false,
+
+          // Should come from the repo settings
+          canDelete: true,
+          canSave: true,
+          canEdit: true,
+
+          // Includes additional k8s metadata
+          k8s: dryRun.metadata,
+
+          // lookup info
+          provisioning: {
+            file: url,
+            ref: ref,
+            repo: repo,
+          },
```

Comment:
@ryantxu are these still needed?

Reply:
🤔  we could be able to remove the provisioning object.  I think we get the same info elsewhere now.  But I would want to try in our full feature branch fist

Reply:
What about `canDelete`, etc? Those are supposed to come from the repo config anyways?

Reply:
Tested in the git-sync branch and it doesn't seem to be used anywhere, also don't get any matches in the code.